docs/tasks/administer-cluster/cloud-controller-manager-deployment-example.yaml

```diff
@@ -0,0 +1,59 @@
+# This is an example of how to setup cloud-controller-manager as a Deployment in your cluster.
+# Note that this Deployment is unlikely to work straight out of the box for your cloud, this is
+# meant to be a guideline.
+
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: cloud-controller-manager
+  namespace: kube-system
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: system:cloud-controller-manager
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+- kind: ServiceAccount
+  name: cloud-controller-manager
+  namespace: kube-system
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    app: cloud-controller-manager
+  annotations:
+    scheduler.alpha.kubernetes.io/critical-pod: ''
+  name: cloud-controller-manager
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      app: cloud-controller-manager
+  template:
+    metadata:
+      labels:
+        app: cloud-controller-manager
+    spec:
+      serviceAccountName: cloud-controller-manager
+      containers:
```

Comment:
Should this also use the node selector to force it to run on the masters and the taint?

Reply:
I don't think so, if you wanted that you should use daemonset

Reply:
From what I've seen it's fairly common to limit control plane components to only master nodes regardless if they are a daemonset or deployment. [Bootkube](https://github.com/kubernetes-incubator/bootkube/blob/4dd7a4a575d0a0a4f24f2452180c4416c52f842a/pkg/asset/internal/templates.go#L419) does it and I _think_ kubeadm does as well.

Reply:
Good point, though I think it serves as a good example then to leave this as a Deployment on non-master nodes since not everyone has their master nodes as schueable. You can also use the Daemonset example to set the node roles and tolerations yourself since these examples aren't really meant to work out of the box anyways, more as guidelines.